model/patch/intent_registry.go

```diff
@@ -0,0 +1,38 @@
+package patch
+
+import "sync"
+
+var intentFactoryRegistry *patchIntentFactoryRegistry
+
+type patchIntentFactory func() Intent
+type patchIntentFactoryRegistry struct {
+	r map[string]patchIntentFactory
+	m sync.RWMutex
+}
+
+func (r *patchIntentFactoryRegistry) get(intentType string) (patchIntentFactory, bool) {
+	r.m.RLock()
+	defer r.m.RUnlock()
```

Comment:
This might be a dumb question, but why is the lock necessary here?

Reply:
It's not, I think strictly, but I think maps in global variables should have mutexs just to prevent the accretion of race conditions, as we might do more operations with this registry.

We're definitely going to see concurrent reads.

Reply:
Not a prob then, I just thought there was something I was missing.